app/routines/generate_reset_code.rb

```diff
@@ -0,0 +1,18 @@
+class GenerateResetCode
+
+  lev_routine
+
+  protected
+
+  def exec(identity, expiration_period = ResetCode::DEFAULT_EXPIRATION_PERIOD)
+    identity.reset_code ||= identity.build_reset_code
```

Comment:
The identity should already be persisted when you pass it to this routine. I guess I should add that to the comments.